blas/tpls/KokkosBlas1_nrm1_tpl_spec_decl.hpp

```diff
@@ -39,161 +39,65 @@ inline void nrm1_print_specialization() {
 namespace KokkosBlas {
 namespace Impl {
 
-#define KOKKOSBLAS1_DNRM1_TPL_SPEC_DECL_BLAS(LAYOUT, MEMSPACE, ETI_SPEC_AVAIL) \
+#define KOKKOSBLAS1_NRM1_TPL_SPEC_DECL_BLAS(SCALAR, LAYOUT, MEMSPACE,          \
+                                            ETI_SPEC_AVAIL)                    \
   template <class ExecSpace>                                                   \
   struct Nrm1<                                                                 \
       ExecSpace,                                                               \
-      Kokkos::View<double, LAYOUT, Kokkos::HostSpace,                          \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
-      Kokkos::View<const double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<typename Kokkos::ArithTraits<SCALAR>::mag_type, LAYOUT,     \
+                   Kokkos::HostSpace,                                          \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      Kokkos::View<const SCALAR*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
       1, true, ETI_SPEC_AVAIL> {                                               \
-    typedef Kokkos::View<double, LAYOUT, Kokkos::HostSpace,                    \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        RV;                                                                    \
-    typedef Kokkos::View<const double*, LAYOUT,                                \
-                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        XV;                                                                    \
-    typedef typename XV::size_type size_type;                                  \
+    using mag_type  = typename Kokkos::ArithTraits<SCALAR>::mag_type;          \
+    using RV        = Kokkos::View<mag_type, LAYOUT, Kokkos::HostSpace,        \
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>;   \
```

Comment:
I feel like the auto-format checker isn't running on these files - shouldn't  the backslashes all be in the same column?

Reply:
Yeah, but then I distinctly recall running clang-format on that file so I am not too sure why it decided to offset that backslash strangely? There might be a small bug in clang-format? This could also be due to a render of tab vs. space by github?